src/en/koharu/src/eu/kanade/tachiyomi/extension/en/koharu/Koharu.kt

```diff
@@ -0,0 +1,279 @@
+package eu.kanade.tachiyomi.extension.en.koharu
+
+import android.app.Application
+import android.content.SharedPreferences
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+import kotlin.reflect.KProperty1
+
+class Koharu : HttpSource(), ConfigurableSource {
+    override val name = "Koharu"
+
+    override val baseUrl = "https://koharu.to"
+
+    private val apiUrl = baseUrl.replace("://", "://api.")
+
+    private val apiBooksUrl = "$apiUrl/books"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(3, 1, TimeUnit.SECONDS)
+        .build()
+
+    private val json = Json { ignoreUnknownKeys = true }
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private var quality = preferences.getString(PREF_IMAGERES, "0")!!

```

Comment:
```suggestion
    private fun quality() = preferences.getString(PREF_IMAGERES, "0")!!
```

also set a resampled quality as default, which they use on the site

Reply:
Hello, I can't seem to find any resampled resolution in their API json. Should I just default the value to just 1280?

Reply:
whichever quality they use by default on the site, i know they dont use original